### PR TITLE
[FIXED JENKINS-27256] Dispatcher for getX(long) model object methods

### PR DESCRIPTION
**Pull request by tfennelly**
See [JENKINS-27256](https://issues.jenkins-ci.org/browse/JENKINS-27256).

Added dispatcher for `getX(long)` style model methods. Already tested manually but will add a functional test in Jenkins core test harness (yeah, should have done that first etc etc).

@reviewbybees

### CODE REVIEW

File: core/src/main/java/org/kohsuke/stapler/TokenList.java [hunk 1]
```diff
@@ -86,12 +86,20 @@ public String prev() {
         return tokens[--idx];
     }
     public int nextAsInt() throws NumberFormatException {
+        long l = nextAsLong();
+        if (l > Integer.MAX_VALUE) {
+            throw new NumberFormatException(String.format("Token '%d' cannot be interpreted as an integer as it's value is greater than %d.", l, Integer.MAX_VALUE));
```
Comment (jglick) [inline core/src/main/java/org/kohsuke/stapler/TokenList.java]: `its`

File: core/src/main/java/org/kohsuke/stapler/TokenList.java [hunk 2]
```diff
@@ -86,12 +86,20 @@ public String prev() {
         return tokens[--idx];
     }
     public int nextAsInt() throws NumberFormatException {
+        long l = nextAsLong();
+        if (l > Integer.MAX_VALUE) {
```
Comment (jglick) [inline core/src/main/java/org/kohsuke/stapler/TokenList.java]: What about `l < Integer.MIN_VALUE`?

Comment (tfennelly) [inline core/src/main/java/org/kohsuke/stapler/TokenList.java]: Good point ... thanks

**Comment by jglick** [comment]
:+1:

**Comment by jtnord** [comment]
:-1: 
"l" (ell) as a variable name shouldn't be used.
return l can be read as return ell or return one and its not obvious which it is and is hard to spot accidental typos.

**Comment by tfennelly** [comment]
Are we happy now James? ffs

**Comment by jtnord** [comment]
:+1: happy chappy

**Comment by daniel-beck** [comment]
https://github.com/stapler/stapler/blob/master/src/site/resources/reference.html#L160 needs to be updated.

**Pull request merged**